howso/residuals.amlg

```diff
@@ -431,6 +431,21 @@
 		;keep a copy of all originally specified features
 		(assign (assoc case_features features))
 
+		;if 'case_ids' are not set, then use 'action_condition_filter_query' to get the case ids of the action set, otherwise
+		;flag that 'case_ids' were provided so 'action_condition_filter_query' is ignored.
+		(if (= (size case_ids) 0)
+			(assign (assoc
+				case_ids
+					(call !GetCasesByCondition (assoc
+						condition_filter_query action_condition_filter_query
+						num_cases num_samples
+					))
+			))
```

Comment:
Not sure if it needs to happen here, or already happens somewhere else. But a consideration for Robust Residuals is that the total number of cases needs to be fairly large. The reasoning being that we need multiple predicitions with/without each feature. Say 12 cases match the query, this is not enough to stably compute robust residuals, so we should sample from those 12 with replacement